scripts/alerts-during-perf-testing.sh

```diff
@@ -0,0 +1,68 @@
+#
+# PREREQUISITES
+# - jq
+# - oc (logged in at the cmd line in order to get the bearer token)
+# VARIABLES
+TOKEN=$(oc whoami --show-token)
+MONITORING_ROUTE=$(echo "https://$(oc get route prometheus-route -n redhat-rhmi-middleware-monitoring-operator -o=jsonpath='{.spec.host}')")/api/v1/alerts
+
+# remove tmp files on ctrl-c
+trap "rm tmp-alert-firing-during-perf-testing-report.csv tmp-alert-pending-during-perf-testing-report.csv" EXIT
+
+# function to check if there are no alerts firing bar deadmansnitch
+function CHECK_NO_ALERTS(){
+    if [[ $(curl -s -H "Authorization: Bearer $TOKEN" $MONITORING_ROUTE | jq -r '.data.alerts[]| select(.state=="firing") | [.labels.alertname, .state, .activeAt ] | @csv' | wc -l) == 1 ]] ; then
+      echo Only alert firing is DeadMansSwitch
+      date
+    fi
+}
+
+# If no args are passed in then run for all products
+if (( $# == 0 )); then
+  while :; do
+    # generate a report
+    curl -s -H "Authorization: Bearer $TOKEN" $MONITORING_ROUTE \
+    | jq -r '.data.alerts[]| select(.state=="pending") | [.labels.alertname, .state, .activeAt ] | @csv'>> tmp-alert-pending-during-perf-testing-report.csv
+
+    curl -s -H "Authorization: Bearer $TOKEN" $MONITORING_ROUTE \
+    | jq -r '.data.alerts[]| select(.state=="firing") | [.labels.alertname, .state, .activeAt ] | @csv'>> tmp-alert-firing-during-perf-testing-report.csv
+
+    # sort command to remove duplicate alert
+    sort -t',' -k 1,1 -u tmp-alert-firing-during-perf-testing-report.csv > alert-firing-during-perf-testing-report.csv
+    #sort -t, -k1 -u alert-firing-during-perf-testing-report.csv > alert-firing-during-perf-testing-report.csv
+    sort -t, -k1 -u tmp-alert-pending-during-perf-testing-report.csv > alert-pending-during-perf-testing-report.csv
+    CHECK_NO_ALERTS
+  done
+fi
+
+# Check the alerts in product namespaces
+PRODUCT_NAME=`echo $1 | grep "3scale\|user-sso\|rhsso\|marin3r"`
+
+if [ -z "$PRODUCT_NAME" ]; then
+  echo Error command line either no args or args excepted for individual products alerts listed below
+  echo - "3scale"
+  echo - "user-sso"
+  echo - "rhsso"
+  echo - "marin3r"
+else
+  echo "Check $1 product namespace and operator namespace for alerts"
+  while :; do
+    curl -s -H "Authorization: Bearer $TOKEN" $MONITORING_ROUTE \
+    | jq -r '.data.alerts[]| select(.state=="pending" or .labels.namespace=="redhat-rhmi-'$PRODUCT_NAME'" or .labels.namespace=="redhat-rhmi-'$PRODUCT_NAME'-operator") | [.labels.alertname, .labels.namespace, .state, .activeAt ] | @csv'>> tmp-alert-pending-during-perf-testing-report.csv
+
+    curl -s -H "Authorization: Bearer $TOKEN" $MONITORING_ROUTE \
+    | jq -r '.data.alerts[]| select(.state=="firing" or .labels.namespace=="redhat-rhmi-'$PRODUCT_NAME'" or .labels.namespace=="redhat-rhmi-'$PRODUCT_NAME'-operator") | [.labels.alertname, .labels.namespace, .state, .activeAt ] | @csv'>> tmp-alert-firing-during-perf-testing-report.csv
```

Comment:
These filters don't seem to work for me. From the output (see below), when I scaled down CRO, it looks like that only CRO has the label `namespace`, but 3scale, rhsso, marin3r and user-sso have only a label `productName`.

<details>
  <summary>Click to see the output</summary>
  
```
curl -s -H 'Authorization: Bearer <token>' https://prometheus-route-redhat-rhmi-middleware-monitoring-operator.apps.test-multiaz-ps.k5b2.s1.devshift.org/api/v1/alerts | jq -r '.data.alerts[]| select(.state=="firing") | [.labels]'
[
  {
    "alertname": "Threescale-Redis-RhmiRedisResourceStatusPhasePending",
    "productName": "3scale",
    "severity": "warning"
  }
]
[
  {
    "alertname": "Threescale-Backend-Redis-RhmiRedisCacheUnavailable",
    "productName": "3scale",
    "severity": "critical"
  }
]
[
  {
    "alertname": "Threescale-Redis-RhmiRedisCacheUnavailable",
    "productName": "3scale",
    "severity": "critical"
  }
]
[
  {
    "alertname": "Ratelimit-Service-Redis-Managed-ApiRedisCacheUnavailable",
    "productName": "marin3r",
    "severity": "critical"
  }
]
[
  {
    "alertname": "DeadMansSwitch",
    "severity": "none"
  }
]
[
  {
    "alertname": "Rhsso-Postgres-Managed-ApiPostgresConnectionFailed",
    "productName": "rhsso",
    "severity": "critical"
  }
]
[
  {
    "alertname": "Threescale-Postgres-Managed-ApiPostgresResourceStatusPhasePending",
    "productName": "3scale",
    "severity": "warning"
  }
]
[
  {
    "alertname": "Rhsso-Postgres-Managed-ApiPostgresResourceStatusPhasePending",
    "productName": "rhsso",
    "severity": "warning"
  }
]
[
  {
    "alertname": "Threescale-Postgres-Managed-ApiPostgresConnectionFailed",
    "productName": "3scale",
    "severity": "critical"
  }
]
[
  {
    "alertname": "Ratelimit-Service-Redis-Managed-ApiRedisResourceStatusPhasePending",
    "productName": "marin3r",
    "severity": "warning"
  }
]
[
  {
    "alertname": "Threescale-Postgres-Managed-ApiPostgresInstanceUnavailable",
    "productName": "3scale",
    "severity": "critical"
  }
]
[
  {
    "alertname": "Rhssouser-Postgres-Managed-ApiPostgresConnectionFailed",
    "productName": "user-sso",
    "severity": "critical"
  }
]
[
  {
    "alertname": "Rhssouser-Postgres-Managed-ApiPostgresResourceStatusPhasePending",
    "productName": "user-sso",
    "severity": "warning"
  }
]
[
  {
    "alertname": "Threescale-Backend-Redis-RhmiRedisResourceStatusPhasePending",
    "productName": "3scale",
    "severity": "warning"
  }
]
[
  {
    "alertname": "RHMICloudResourceOperatorMetricsServiceEndpointDown",
    "endpoint": "cloud-resource-operator-metrics",
    "instance": "10.11.90.8:8443",
    "job": "kube-state-metrics",
    "namespace": "redhat-rhmi-cloud-resources-operator",
    "pod": "kube-state-metrics-5fc5b5794f-9bx77",
    "prometheus": "openshift-monitoring/k8s",
    "service": "kube-state-metrics",
    "severity": "critical"
  }
]
[
  {
    "alertname": "Rhsso-Postgres-Managed-ApiPostgresInstanceUnavailable",
    "productName": "rhsso",
    "severity": "critical"
  }
]
[
  {
    "alertname": "Rhssouser-Postgres-Managed-ApiPostgresInstanceUnavailable",
    "productName": "user-sso",
    "severity": "critical"
  }
]
```
</details>

Maybe this could work?

I.e.
```suggestion
    curl -s -H "Authorization: Bearer $TOKEN" $MONITORING_ROUTE \
    | jq -r '.data.alerts[]| select(.state=="firing" and .labels.productName=="'$PRODUCT_NAME'") | [.labels.alertname, .labels.productName, .state, .activeAt ] | @csv'>> tmp-alert-pending-during-perf-testing-report.csv

    curl -s -H "Authorization: Bearer $TOKEN" $MONITORING_ROUTE \
    | jq -r '.data.alerts[]| select(.state=="pending" and .labels.productName=="'$PRODUCT_NAME'") | [.labels.alertname, .labels.productName, .state, .activeAt ] | @csv'>> tmp-alert-firing-during-perf-testing-report.csv
```

Reply:
Unfortunately there is no label that is common to all alerts , see https://issues.redhat.com/browse/MGDAPI-172?focusedCommentId=15320748&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15320748, only cro uses productName, most alerts are covered by namespaces but not all. May not be worth having this filter. I could add an extra conditional to cover cro
```suggestion
    curl -s -H "Authorization: Bearer $TOKEN" $MONITORING_ROUTE \
    | jq -r '.data.alerts[]| select((.state=="pending") and (.labels.namespace=="redhat-rhmi-'$PRODUCT_NAME'" or .labels.namespace=="redhat-rhmi-'$PRODUCT_NAME'-operator" or .labels.productName=='$PRODUCT_NAME'")) | [.labels.alertname, .labels.namespace, .state, .activeAt ] | @csv'>> tmp-alert-pending-during-perf-testing-report.csv

    curl -s -H "Authorization: Bearer $TOKEN" $MONITORING_ROUTE \
    | jq -r '.data.alerts[]| select((.state=="firing") and (.labels.namespace=="redhat-rhmi-'$PRODUCT_NAME'" or .labels.namespace=="redhat-rhmi-'$PRODUCT_NAME'-operator" or .labels.productName=='$PRODUCT_NAME'")) | [.labels.alertname, .labels.namespace, .state, .activeAt ] | @csv'>> tmp-alert-firing-during-perf-testing-report.csv
``` but alerts like KeycloakInstanceNotAvailable will not be covered

Reply:
e.g. of labels available when user-sso is down
```bash
curl -s -H "Authorization: Bearer $(oc whoami --show-token)" \
    $(echo "https://$(oc get route prometheus-route -n redhat-rhmi-middleware-monitoring-operator -o=jsonpath='{.spec.host}')")/api/v1/alerts \
    | jq -r '.data.alerts[]| select(.state=="firing") | .labels'        
```
```json
{
  "alertname": "KeycloakInstanceNotAvailable",
  "severity": "critical"
}
{
  "alertname": "RHMIUserRhssoKeycloakServiceEndpointDown",
  "endpoint": "keycloak",
  "instance": "10.130.6.7:8443",
  "job": "kube-state-metrics",
  "namespace": "redhat-rhmi-user-sso",
  "pod": "kube-state-metrics-5fc5b5794f-294cq",
  "prometheus": "openshift-monitoring/k8s",
  "service": "kube-state-metrics",
  "severity": "critical"
}
{
  "alertname": "RHMIUserRhssoKeycloakDiscoveryServiceEndpointDown",
  "endpoint": "keycloak-discovery",
  "instance": "10.130.6.7:8443",
  "job": "kube-state-metrics",
  "namespace": "redhat-rhmi-user-sso",
  "pod": "kube-state-metrics-5fc5b5794f-294cq",
  "prometheus": "openshift-monitoring/k8s",
  "service": "kube-state-metrics",
  "severity": "critical"
}
{
  "alertname": "RHMIUserRhssoKeycloakOperatorMetricsServiceEndpointDown",
  "endpoint": "keycloak-operator-metrics",
  "instance": "10.130.6.7:8443",
  "job": "kube-state-metrics",
  "namespace": "redhat-rhmi-user-sso-operator",
  "pod": "kube-state-metrics-5fc5b5794f-294cq",
  "prometheus": "openshift-monitoring/k8s",
  "service": "kube-state-metrics",
  "severity": "critical"
}
{
  "alertname": "DeadMansSwitch",
  "severity": "none"
}
```

Reply:
looks like all the keycloak operator alerts don't have any labels, I get why they don't have namespace or productName labels as that wouldn't suit Keycloak beyond RHOAM. Hard to know which will be hit during performance and load testing might be able to use regex to search for alert.name for something like `Keycloak(.+)$`